bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BDecimal.java

```diff
@@ -94,6 +131,203 @@ public BValue copy(Map<BValue, BValue> refs) {
         return this;
     }
 
+    //========================= Mathematical operations supported ===============================
+
+    public BDecimal add(BDecimal augend) {
+        switch (this.valueKind) {
+            case ZERO:
+                return augend;
+
+            case POSITIVE_INFINITY:
+                if (augend.valueKind == DecimalValueKind.NEGATIVE_INFINITY ||
+                        augend.valueKind == DecimalValueKind.NOT_A_NUMBER) {
+                    return NaN;
+                }
+                return POSITIVE_INF;
+
+            case NEGATIVE_INFINITY:
+                if (augend.valueKind == DecimalValueKind.POSITIVE_INFINITY ||
+                        augend.valueKind == DecimalValueKind.NOT_A_NUMBER) {
+                    return NaN;
+                }
+                return NEGATIVE_INF;
+
+            case NOT_A_NUMBER:
+                return NaN;
+
+            default:
+                if (augend.valueKind == DecimalValueKind.ZERO || augend.valueKind == DecimalValueKind.OTHER) {
```

Comment:
if DecimalValueKind.ZERO we can return new BDecimal(this.decimalValue(), MathContext.DECIMAL128) without calling add? similarly in subtract as well

Reply:
+1. Will modify accordingly.